tests/TypeReconciliation/EmptyTest.php

```diff
@@ -626,6 +634,11 @@ function test(string $s): void {
                     '$x===' => 'true',
                 ],
             ],
+            'emptyArrayFetch' => [
+                'code' => '<?php
+                    /** @var array<true> $a */
+                    if (empty($a["a"])) {}',
+            ],
```

Comment:
I'd suggest adding another test like this:
```php
/** @var array<true> $a*/;
assert(isset($a["a"]));
if (empty($a["a"])) {} // it should report `TypeDoesNotContainType` (or paradox) here
```

Reply:
The error I get is `DocblockTypeContradiction`, which...doesn't seem 100% correct, since the contradiction doesn't come from the docblock type, but rather the assertion. But I suspect this is a separate issue?